### DISCUSSION TITLE
Access localStorage

### DISCUSSION THREAD

**Discussion by cemoktra** [alephjs/aleph.js#142] [Q&A]
Hi i have a problem accesing the localStorage:

```import 'https://deno.land/x/localstorage/mod.ts'

export default function Home() {
  const {localStorage} = window
  localStorage.setItem("test", 123)
```
This results in:
`Uncaught (in promise) ReferenceError: Deno is not defined`

As far as i understood `useDeno` is made for this, so i changed it to: 
```import 'https://deno.land/x/localstorage/mod.ts'

export default function Home() {
  const {localStorage} = useDeno(() => window)
  localStorage.setItem("test", 123)
```

and get:
`500 - Access to "location", run again with --location .`

What do i understand wrong?

**Comment by ije** [comment]
@cemoktra currently deno don't support `localStorage` i think, please check https://github.com/denoland/deno/issues/1657

**Reply by ije** [reply]
you can use it in the `useEffect` hook safely or in spa mode

**Reply by cemoktra** [reply]
@ije therefore i use https://deno.land/x/localstorage@0.1.0

but it's still not clear to me how to use it? Where is useEffect defined?

**Reply by cemoktra** [reply]
Actually i could live with a different solution, i just want to store a session token somehow

**Reply by ije** [reply]
useDeno is a ssr solution, you can not write the return value, you may need the API instead.

```js
const value = useDeno(() => window.localStorage.getItem('hello'))
```

**Reply by FallingSnow** [reply]
@cemoktra useEffect is react hook. https://reactjs.org/docs/hooks-effect.html

You define it inside a react functional component.
